src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -41,6 +41,71 @@ def _record_cancellation_exceptions_to_span(span: Span):
         raise
 
 
+def handle_span_exception(span, exception):
+    if isinstance(span, Span) and span.is_recording():
+        # Record the exception as an event
+        span.record_exception(exception)
+        # Records status as error
+        span.set_status(
+            Status(
+                status_code=StatusCode.ERROR,
+                description=f"{type(exception).__name__}: {str(exception)}",
+            )
+        )
+
+
+def handle_output(span, inputs, output, trace_type):
+    if isinstance(output, (Iterator, AsyncIterator)):
+        # Set should_end to False to delay span end until generator exhaustion, preventing premature span end.
+        setattr(span, "__should_end", False)
+        output = Tracer.pop(output)
+        if isinstance(output, Iterator):
+            return traced_generator(span, inputs, output, trace_type)
+        else:
+            return traced_async_generator(span, inputs, output, trace_type)
+    else:
+        enrich_span_with_trace_type(span, inputs, output, trace_type)
+        span.set_status(StatusCode.OK)
+        return Tracer.pop(output)
+
+
+@contextlib.contextmanager
+def start_as_current_span(
+    tracer: otel_trace.Tracer,
+    name: str,
+    context: Optional[otel_trace.Context] = None,
+    kind: otel_trace.SpanKind = otel_trace.SpanKind.INTERNAL,
+    attributes: types.Attributes = None,
+    links: Optional[Sequence[otel_trace.Link]] = (),
+    start_time: Optional[int] = None,
+    record_exception: bool = True,
+    set_status_on_exception: bool = True,
+):
+    span = None
+    try:
+        with tracer.start_as_current_span(
+            name,
+            context,
+            kind,
+            attributes,
+            links,
+            start_time,
+            record_exception,
+            set_status_on_exception,
+            end_on_exit=False,
+        ) as span:
+            object.__setattr__(span, "__should_end", True)
+            yield span
+
+    except (KeyboardInterrupt, asyncio.CancelledError) as ex:
+        handle_span_exception(span, ex)
```

Comment:
Add comment here to say that these exceptions are not handled by the otel framework?

Reply:
OK